// unenv/runtime/node/util/types/$cloudflare.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isDate = exports.isDataView = exports.isCryptoKey = exports.isBoxedPrimitive = exports.isBooleanObject = exports.isBigUint64Array = exports.isBigIntObject = exports.isBigInt64Array = exports.isAsyncFunction = exports.isArrayBufferView = exports.isArrayBuffer = exports.isArgumentsObject = exports.isAnyArrayBuffer = exports.default = void 0;
Object.defineProperty(exports, "isExternal", {
  enumerable: true,
  get: function () {
    return _index.isExternal;
  }
});
exports.isWeakSet = exports.isWeakMap = exports.isUint8ClampedArray = exports.isUint8Array = exports.isUint32Array = exports.isUint16Array = exports.isTypedArray = exports.isSymbolObject = exports.isStringObject = exports.isSharedArrayBuffer = exports.isSetIterator = exports.isSet = exports.isRegExp = exports.isProxy = exports.isPromise = exports.isNumberObject = exports.isNativeError = exports.isModuleNamespaceObject = exports.isMapIterator = exports.isMap = exports.isKeyObject = exports.isInt8Array = exports.isInt32Array = exports.isInt16Array = exports.isGeneratorObject = exports.isGeneratorFunction = exports.isFloat64Array = exports.isFloat32Array = void 0;
var _index = require("./index.cjs");
const workerdUtil = process.getBuiltinModule("node:util");
const {
  isAnyArrayBuffer,
  isArgumentsObject,
  isArrayBuffer,
  isArrayBufferView,
  isAsyncFunction,
  isBigInt64Array,
  isBigIntObject,
  isBigUint64Array,
  isBooleanObject,
  isBoxedPrimitive,
  isCryptoKey,
  isDataView,
  isDate,
  isFloat32Array,
  isFloat64Array,
  isGeneratorFunction,
  isGeneratorObject,
  isInt16Array,
  isInt32Array,
  isInt8Array,
  isKeyObject,
  isMap,
  isMapIterator,
  isModuleNamespaceObject,
  isNativeError,
  isNumberObject,
  isPromise,
  isProxy,
  isRegExp,
  isSet,
  isSetIterator,
  isSharedArrayBuffer,
  isStringObject,
  isSymbolObject,
  isTypedArray,
  isUint16Array,
  isUint32Array,
  isUint8Array,
  isUint8ClampedArray,
  isWeakMap,
  isWeakSet
} = workerdUtil.types;
exports.isWeakSet = isWeakSet;
exports.isWeakMap = isWeakMap;
exports.isUint8ClampedArray = isUint8ClampedArray;
exports.isUint8Array = isUint8Array;
exports.isUint32Array = isUint32Array;
exports.isUint16Array = isUint16Array;
exports.isTypedArray = isTypedArray;
exports.isSymbolObject = isSymbolObject;
exports.isStringObject = isStringObject;
exports.isSharedArrayBuffer = isSharedArrayBuffer;
exports.isSetIterator = isSetIterator;
exports.isSet = isSet;
exports.isRegExp = isRegExp;
exports.isProxy = isProxy;
exports.isPromise = isPromise;
exports.isNumberObject = isNumberObject;
exports.isNativeError = isNativeError;
exports.isModuleNamespaceObject = isModuleNamespaceObject;
exports.isMapIterator = isMapIterator;
exports.isMap = isMap;
exports.isKeyObject = isKeyObject;
exports.isInt8Array = isInt8Array;
exports.isInt32Array = isInt32Array;
exports.isInt16Array = isInt16Array;
exports.isGeneratorObject = isGeneratorObject;
exports.isGeneratorFunction = isGeneratorFunction;
exports.isFloat64Array = isFloat64Array;
exports.isFloat32Array = isFloat32Array;
exports.isDate = isDate;
exports.isDataView = isDataView;
exports.isCryptoKey = isCryptoKey;
exports.isBoxedPrimitive = isBoxedPrimitive;
exports.isBooleanObject = isBooleanObject;
exports.isBigUint64Array = isBigUint64Array;
exports.isBigIntObject = isBigIntObject;
exports.isBigInt64Array = isBigInt64Array;
exports.isAsyncFunction = isAsyncFunction;
exports.isArrayBufferView = isArrayBufferView;
exports.isArrayBuffer = isArrayBuffer;
exports.isArgumentsObject = isArgumentsObject;
exports.isAnyArrayBuffer = isAnyArrayBuffer;
module.exports = {
  /**
   * manually unroll unenv-polyfilled-symbols to make it tree-shakeable
   */
  isExternal: _index.isExternal,
  /**
   * manually unroll workerd-polyfilled-symbols to make it tree-shakeable
   */
  isAnyArrayBuffer,
  isArgumentsObject,
  isArrayBuffer,
  isArrayBufferView,
  isAsyncFunction,
  isBigInt64Array,
  // @ts-expect-error undocumented public API
  isBigIntObject,
  isBigUint64Array,
  isBooleanObject,
  isBoxedPrimitive,
  isCryptoKey,
  isDataView,
  isDate,
  isFloat32Array,
  isFloat64Array,
  isGeneratorFunction,
  isGeneratorObject,
  isInt16Array,
  isInt32Array,
  isInt8Array,
  isKeyObject,
  isMap,
  isMapIterator,
  isModuleNamespaceObject,
  isNativeError,
  isNumberObject,
  isPromise,
  isProxy,
  isRegExp,
  isSet,
  isSetIterator,
  isSharedArrayBuffer,
  isStringObject,
  isSymbolObject,
  isTypedArray,
  isUint16Array,
  isUint32Array,
  isUint8Array,
  isUint8ClampedArray,
  isWeakMap,
  isWeakSet
};
