// unenv/runtime/node/buffer/$cloudflare.mjs
export {
  Blob,
  File,
  INSPECT_MAX_BYTES,
  atob,
  btoa,
  isAscii,
  isUtf8,
  resolveObjectURL,
  transcode
} from "./index.mjs";
import {
  Blob as Blob2,
  File as File2,
  INSPECT_MAX_BYTES as INSPECT_MAX_BYTES2,
  atob as atob2,
  btoa as btoa2,
  isAscii as isAscii2,
  isUtf8 as isUtf82,
  resolveObjectURL as resolveObjectURL2,
  transcode as transcode2
} from "./index.mjs";
const workerdBuffer = process.getBuiltinModule("node:buffer");
export const { Buffer, SlowBuffer, constants, kMaxLength, kStringMaxLength } = workerdBuffer;
export default {
  /**
   * manually unroll unenv-polyfilled-symbols to make it tree-shakeable
   */
  Blob: Blob2,
  File: File2,
  INSPECT_MAX_BYTES: INSPECT_MAX_BYTES2,
  atob: atob2,
  btoa: btoa2,
  isAscii: isAscii2,
  isUtf8: isUtf82,
  resolveObjectURL: resolveObjectURL2,
  transcode: transcode2,
  /**
   * manually unroll workerd-polyfilled-symbols to make it tree-shakeable
   */
  Buffer,
  SlowBuffer,
  constants,
  kMaxLength,
  kStringMaxLength
};
