// unenv/runtime/node/timers/index.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.unenroll = exports.setTimeout = exports.setInterval = exports.setImmediate = exports.promises = exports.enroll = exports.default = exports.clearTimeout = exports.clearInterval = exports.clearImmediate = exports.active = exports._unrefActive = void 0;
var _utils = require("../../_internal/utils.cjs");
var _noop = _interopRequireDefault(require("../../mock/noop.cjs"));
var _promises2 = _interopRequireWildcard(require("./promises/index.cjs"));
var _promises = _promises2;
exports.promises = _promises2;
var _setTimeout = require("./internal/set-timeout.cjs");
var _setImmediate = require("./internal/set-immediate.cjs");
var _setInterval = require("./internal/set-interval.cjs");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const clearImmediate = exports.clearImmediate = globalThis.clearImmediate || _noop.default;
const clearInterval = exports.clearInterval = globalThis.clearInterval || _noop.default;
const clearTimeout = exports.clearTimeout = globalThis.clearTimeout || _noop.default;
const setImmediate = exports.setImmediate = globalThis.setImmediate || _setImmediate.setImmediateFallback;
const setTimeout = exports.setTimeout = globalThis.setTimeout || _setTimeout.setTimeoutFallback;
const setInterval = exports.setInterval = globalThis.setInterval || _setInterval.setIntervalFallback;
const active = exports.active = (0, _utils.notImplemented)("timers.active");
const _unrefActive = exports._unrefActive = (0, _utils.notImplemented)("timers._unrefActive");
const enroll = exports.enroll = (0, _utils.notImplemented)("timers.enroll");
const unenroll = exports.unenroll = (0, _utils.notImplemented)("timers.unenroll");
module.exports = {
  _unrefActive,
  active,
  clearImmediate,
  clearInterval,
  clearTimeout,
  enroll,
  promises: _promises2.default,
  setImmediate,
  setInterval,
  setTimeout,
  unenroll
};
