// unenv/runtime/node/child_process/index.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.spawnSync = exports.spawn = exports.fork = exports.execSync = exports.execFileSync = exports.execFile = exports.exec = exports.default = exports._forkChild = exports.ChildProcess = void 0;
var _utils = require("../../_internal/utils.cjs");
const ChildProcess = exports.ChildProcess = (0, _utils.notImplementedClass)("child_process.ChildProcess");
const _forkChild = exports._forkChild = (0, _utils.notImplemented)("child_process.ChildProcess");
const exec = exports.exec = (0, _utils.notImplemented)("child_process.exec");
const execFile = exports.execFile = (0, _utils.notImplemented)("child_process.execFile");
const execFileSync = exports.execFileSync = (0, _utils.notImplemented)("child_process.execFileSync");
const execSync = exports.execSync = (0, _utils.notImplemented)("child_process.execSyn");
const fork = exports.fork = (0, _utils.notImplemented)("child_process.fork");
const spawn = exports.spawn = (0, _utils.notImplemented)("child_process.spawn");
const spawnSync = exports.spawnSync = (0, _utils.notImplemented)("child_process.spawnSync");
module.exports = {
  ChildProcess,
  _forkChild,
  exec,
  execFile,
  execFileSync,
  execSync,
  fork,
  spawn,
  spawnSync
};
