// unenv/runtime/web/performance/_observer.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._PerformanceObserverEntryList = exports._PerformanceObserver = exports.PerformanceObserverEntryList = exports.PerformanceObserver = void 0;
var _utils = require("../../_internal/utils.cjs");
var _entry = require("./_entry.cjs");
class _PerformanceObserver {
  __unenv__ = true;
  static supportedEntryTypes = _entry._supportedEntryTypes;
  _callback = null;
  constructor(callback) {
    this._callback = callback;
  }
  takeRecords() {
    return [];
  }
  disconnect() {
    throw (0, _utils.createNotImplementedError)("PerformanceObserver.disconnect");
  }
  observe(options) {
    throw (0, _utils.createNotImplementedError)("PerformanceObserver.observe");
  }
}
exports._PerformanceObserver = _PerformanceObserver;
const PerformanceObserver = exports.PerformanceObserver = globalThis.PerformanceObserver || _PerformanceObserver;
class _PerformanceObserverEntryList {
  __unenv__ = true;
  getEntries() {
    return [];
  }
  getEntriesByName(_name, _type) {
    return [];
  }
  getEntriesByType(type) {
    return [];
  }
}
exports._PerformanceObserverEntryList = _PerformanceObserverEntryList;
const PerformanceObserverEntryList = exports.PerformanceObserverEntryList = globalThis.PerformanceObserverEntryList || _PerformanceObserverEntryList;
