// unenv/runtime/node/zlib/internal/formats/brotli.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createBrotliDecompress = exports.createBrotliCompress = exports.brotliDecompressSync = exports.brotliDecompress = exports.brotliCompressSync = exports.brotliCompress = exports.BrotliDecompress = exports.BrotliCompress = void 0;
var _utils = require("../../../../_internal/utils.cjs");
var _shared = require("./_shared.cjs");
class BrotliCompress extends _shared.ZlibCompress {
  _format = "brotli";
}
exports.BrotliCompress = BrotliCompress;
const brotliCompress = exports.brotliCompress = (0, _shared.notImplementedCompress)("brotliCompress");
const createBrotliCompress = () => new BrotliCompress();
exports.createBrotliCompress = createBrotliCompress;
const brotliCompressSync = exports.brotliCompressSync = (0, _utils.notImplemented)("zlib.brotliCompressSync");
class BrotliDecompress extends _shared.ZLibDecompress {
  _format = "brotli";
}
exports.BrotliDecompress = BrotliDecompress;
const brotliDecompress = exports.brotliDecompress = (0, _shared.notImplementedCompress)("brotliDecompress");
const createBrotliDecompress = () => new BrotliDecompress();
exports.createBrotliDecompress = createBrotliDecompress;
const brotliDecompressSync = exports.brotliDecompressSync = (0, _utils.notImplemented)("zlib.brotliDecompressSync");
