// unenv/runtime/node/constants/iternal/crypto.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defaultCoreCipherList = exports.defaultCipherList = exports.TLS1_VERSION = exports.TLS1_3_VERSION = exports.TLS1_2_VERSION = exports.TLS1_1_VERSION = exports.SSL_OP_TLS_ROLLBACK_BUG = exports.SSL_OP_PRIORITIZE_CHACHA = exports.SSL_OP_NO_TLSv1_3 = exports.SSL_OP_NO_TLSv1_2 = exports.SSL_OP_NO_TLSv1_1 = exports.SSL_OP_NO_TLSv1 = exports.SSL_OP_NO_TICKET = exports.SSL_OP_NO_SSLv3 = exports.SSL_OP_NO_SSLv2 = exports.SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION = exports.SSL_OP_NO_RENEGOTIATION = exports.SSL_OP_NO_QUERY_MTU = exports.SSL_OP_NO_ENCRYPT_THEN_MAC = exports.SSL_OP_NO_COMPRESSION = exports.SSL_OP_LEGACY_SERVER_CONNECT = exports.SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS = exports.SSL_OP_CRYPTOPRO_TLSEXT_BUG = exports.SSL_OP_COOKIE_EXCHANGE = exports.SSL_OP_CISCO_ANYCONNECT = exports.SSL_OP_CIPHER_SERVER_PREFERENCE = exports.SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION = exports.SSL_OP_ALLOW_NO_DHE_KEX = exports.SSL_OP_ALL = exports.RSA_X931_PADDING = exports.RSA_SSLV23_PADDING = exports.RSA_PSS_SALTLEN_MAX_SIGN = exports.RSA_PSS_SALTLEN_DIGEST = exports.RSA_PSS_SALTLEN_AUTO = exports.RSA_PKCS1_PSS_PADDING = exports.RSA_PKCS1_PADDING = exports.RSA_PKCS1_OAEP_PADDING = exports.RSA_NO_PADDING = exports.POINT_CONVERSION_UNCOMPRESSED = exports.POINT_CONVERSION_HYBRID = exports.POINT_CONVERSION_COMPRESSED = exports.OPENSSL_VERSION_NUMBER = exports.ENGINE_METHOD_RSA = exports.ENGINE_METHOD_RAND = exports.ENGINE_METHOD_PKEY_METHS = exports.ENGINE_METHOD_PKEY_ASN1_METHS = exports.ENGINE_METHOD_NONE = exports.ENGINE_METHOD_EC = exports.ENGINE_METHOD_DSA = exports.ENGINE_METHOD_DIGESTS = exports.ENGINE_METHOD_DH = exports.ENGINE_METHOD_CIPHERS = exports.ENGINE_METHOD_ALL = exports.DH_UNABLE_TO_CHECK_GENERATOR = exports.DH_NOT_SUITABLE_GENERATOR = exports.DH_CHECK_P_NOT_SAFE_PRIME = exports.DH_CHECK_P_NOT_PRIME = void 0;
const OPENSSL_VERSION_NUMBER = exports.OPENSSL_VERSION_NUMBER = 0;
const SSL_OP_ALL = exports.SSL_OP_ALL = 2147485780;
const SSL_OP_ALLOW_NO_DHE_KEX = exports.SSL_OP_ALLOW_NO_DHE_KEX = 1024;
const SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION = exports.SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION = 262144;
const SSL_OP_CIPHER_SERVER_PREFERENCE = exports.SSL_OP_CIPHER_SERVER_PREFERENCE = 4194304;
const SSL_OP_CISCO_ANYCONNECT = exports.SSL_OP_CISCO_ANYCONNECT = 32768;
const SSL_OP_COOKIE_EXCHANGE = exports.SSL_OP_COOKIE_EXCHANGE = 8192;
const SSL_OP_CRYPTOPRO_TLSEXT_BUG = exports.SSL_OP_CRYPTOPRO_TLSEXT_BUG = 2147483648;
const SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS = exports.SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS = 2048;
const SSL_OP_LEGACY_SERVER_CONNECT = exports.SSL_OP_LEGACY_SERVER_CONNECT = 4;
const SSL_OP_NO_COMPRESSION = exports.SSL_OP_NO_COMPRESSION = 131072;
const SSL_OP_NO_ENCRYPT_THEN_MAC = exports.SSL_OP_NO_ENCRYPT_THEN_MAC = 524288;
const SSL_OP_NO_QUERY_MTU = exports.SSL_OP_NO_QUERY_MTU = 4096;
const SSL_OP_NO_RENEGOTIATION = exports.SSL_OP_NO_RENEGOTIATION = 1073741824;
const SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION = exports.SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION = 65536;
const SSL_OP_NO_SSLv2 = exports.SSL_OP_NO_SSLv2 = 0;
const SSL_OP_NO_SSLv3 = exports.SSL_OP_NO_SSLv3 = 33554432;
const SSL_OP_NO_TICKET = exports.SSL_OP_NO_TICKET = 16384;
const SSL_OP_NO_TLSv1 = exports.SSL_OP_NO_TLSv1 = 67108864;
const SSL_OP_NO_TLSv1_1 = exports.SSL_OP_NO_TLSv1_1 = 268435456;
const SSL_OP_NO_TLSv1_2 = exports.SSL_OP_NO_TLSv1_2 = 134217728;
const SSL_OP_NO_TLSv1_3 = exports.SSL_OP_NO_TLSv1_3 = 536870912;
const SSL_OP_PRIORITIZE_CHACHA = exports.SSL_OP_PRIORITIZE_CHACHA = 2097152;
const SSL_OP_TLS_ROLLBACK_BUG = exports.SSL_OP_TLS_ROLLBACK_BUG = 8388608;
const ENGINE_METHOD_RSA = exports.ENGINE_METHOD_RSA = 1;
const ENGINE_METHOD_DSA = exports.ENGINE_METHOD_DSA = 2;
const ENGINE_METHOD_DH = exports.ENGINE_METHOD_DH = 4;
const ENGINE_METHOD_RAND = exports.ENGINE_METHOD_RAND = 8;
const ENGINE_METHOD_EC = exports.ENGINE_METHOD_EC = 2048;
const ENGINE_METHOD_CIPHERS = exports.ENGINE_METHOD_CIPHERS = 64;
const ENGINE_METHOD_DIGESTS = exports.ENGINE_METHOD_DIGESTS = 128;
const ENGINE_METHOD_PKEY_METHS = exports.ENGINE_METHOD_PKEY_METHS = 512;
const ENGINE_METHOD_PKEY_ASN1_METHS = exports.ENGINE_METHOD_PKEY_ASN1_METHS = 1024;
const ENGINE_METHOD_ALL = exports.ENGINE_METHOD_ALL = 65535;
const ENGINE_METHOD_NONE = exports.ENGINE_METHOD_NONE = 0;
const DH_CHECK_P_NOT_SAFE_PRIME = exports.DH_CHECK_P_NOT_SAFE_PRIME = 2;
const DH_CHECK_P_NOT_PRIME = exports.DH_CHECK_P_NOT_PRIME = 1;
const DH_UNABLE_TO_CHECK_GENERATOR = exports.DH_UNABLE_TO_CHECK_GENERATOR = 4;
const DH_NOT_SUITABLE_GENERATOR = exports.DH_NOT_SUITABLE_GENERATOR = 8;
const RSA_PKCS1_PADDING = exports.RSA_PKCS1_PADDING = 1;
const RSA_SSLV23_PADDING = exports.RSA_SSLV23_PADDING = 2;
const RSA_NO_PADDING = exports.RSA_NO_PADDING = 3;
const RSA_PKCS1_OAEP_PADDING = exports.RSA_PKCS1_OAEP_PADDING = 4;
const RSA_X931_PADDING = exports.RSA_X931_PADDING = 5;
const RSA_PKCS1_PSS_PADDING = exports.RSA_PKCS1_PSS_PADDING = 6;
const RSA_PSS_SALTLEN_DIGEST = exports.RSA_PSS_SALTLEN_DIGEST = -1;
const RSA_PSS_SALTLEN_MAX_SIGN = exports.RSA_PSS_SALTLEN_MAX_SIGN = -2;
const RSA_PSS_SALTLEN_AUTO = exports.RSA_PSS_SALTLEN_AUTO = -2;
const defaultCoreCipherList = exports.defaultCoreCipherList = "TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA256:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!SRP:!CAMELLIA";
const TLS1_VERSION = exports.TLS1_VERSION = 0;
const TLS1_1_VERSION = exports.TLS1_1_VERSION = 0;
const TLS1_2_VERSION = exports.TLS1_2_VERSION = 0;
const TLS1_3_VERSION = exports.TLS1_3_VERSION = 0;
const POINT_CONVERSION_COMPRESSED = exports.POINT_CONVERSION_COMPRESSED = 2;
const POINT_CONVERSION_UNCOMPRESSED = exports.POINT_CONVERSION_UNCOMPRESSED = 4;
const POINT_CONVERSION_HYBRID = exports.POINT_CONVERSION_HYBRID = 6;
const defaultCipherList = exports.defaultCipherList = "";
