// unenv/runtime/node/constants/iternal/fs.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.X_OK = exports.W_OK = exports.UV_FS_SYMLINK_JUNCTION = exports.UV_FS_SYMLINK_DIR = exports.UV_FS_O_FILEMAP = exports.UV_FS_COPYFILE_FICLONE_FORCE = exports.UV_FS_COPYFILE_FICLONE = exports.UV_FS_COPYFILE_EXCL = exports.UV_DIRENT_UNKNOWN = exports.UV_DIRENT_SOCKET = exports.UV_DIRENT_LINK = exports.UV_DIRENT_FILE = exports.UV_DIRENT_FIFO = exports.UV_DIRENT_DIR = exports.UV_DIRENT_CHAR = exports.UV_DIRENT_BLOCK = exports.S_IXUSR = exports.S_IXOTH = exports.S_IXGRP = exports.S_IWUSR = exports.S_IWOTH = exports.S_IWGRP = exports.S_IRWXU = exports.S_IRWXO = exports.S_IRWXG = exports.S_IRUSR = exports.S_IROTH = exports.S_IRGRP = exports.S_IFSOCK = exports.S_IFREG = exports.S_IFMT = exports.S_IFLNK = exports.S_IFIFO = exports.S_IFDIR = exports.S_IFCHR = exports.S_IFBLK = exports.R_OK = exports.O_WRONLY = exports.O_TRUNC = exports.O_SYNC = exports.O_SYMLINK = exports.O_RDWR = exports.O_RDONLY = exports.O_NONBLOCK = exports.O_NOFOLLOW = exports.O_NOCTTY = exports.O_NOATIME = exports.O_EXCL = exports.O_DSYNC = exports.O_DIRECTORY = exports.O_DIRECT = exports.O_CREAT = exports.O_APPEND = exports.F_OK = exports.EXTENSIONLESS_FORMAT_WASM = exports.EXTENSIONLESS_FORMAT_JAVASCRIPT = exports.COPYFILE_FICLONE_FORCE = exports.COPYFILE_FICLONE = exports.COPYFILE_EXCL = void 0;
const UV_FS_SYMLINK_DIR = exports.UV_FS_SYMLINK_DIR = 1;
const UV_FS_SYMLINK_JUNCTION = exports.UV_FS_SYMLINK_JUNCTION = 2;
const O_RDONLY = exports.O_RDONLY = 0;
const O_WRONLY = exports.O_WRONLY = 1;
const O_RDWR = exports.O_RDWR = 2;
const UV_DIRENT_UNKNOWN = exports.UV_DIRENT_UNKNOWN = 0;
const UV_DIRENT_FILE = exports.UV_DIRENT_FILE = 1;
const UV_DIRENT_DIR = exports.UV_DIRENT_DIR = 2;
const UV_DIRENT_LINK = exports.UV_DIRENT_LINK = 3;
const UV_DIRENT_FIFO = exports.UV_DIRENT_FIFO = 4;
const UV_DIRENT_SOCKET = exports.UV_DIRENT_SOCKET = 5;
const UV_DIRENT_CHAR = exports.UV_DIRENT_CHAR = 6;
const UV_DIRENT_BLOCK = exports.UV_DIRENT_BLOCK = 7;
const S_IFMT = exports.S_IFMT = 61440;
const S_IFREG = exports.S_IFREG = 32768;
const S_IFDIR = exports.S_IFDIR = 16384;
const S_IFCHR = exports.S_IFCHR = 8192;
const S_IFBLK = exports.S_IFBLK = 24576;
const S_IFIFO = exports.S_IFIFO = 4096;
const S_IFLNK = exports.S_IFLNK = 40960;
const S_IFSOCK = exports.S_IFSOCK = 49152;
const O_CREAT = exports.O_CREAT = 512;
const O_EXCL = exports.O_EXCL = 2048;
const UV_FS_O_FILEMAP = exports.UV_FS_O_FILEMAP = 0;
const O_NOCTTY = exports.O_NOCTTY = 131072;
const O_TRUNC = exports.O_TRUNC = 1024;
const O_APPEND = exports.O_APPEND = 8;
const O_DIRECTORY = exports.O_DIRECTORY = 1048576;
const O_NOFOLLOW = exports.O_NOFOLLOW = 256;
const O_SYNC = exports.O_SYNC = 128;
const O_DSYNC = exports.O_DSYNC = 4194304;
const O_SYMLINK = exports.O_SYMLINK = 2097152;
const O_NONBLOCK = exports.O_NONBLOCK = 4;
const S_IRWXU = exports.S_IRWXU = 448;
const S_IRUSR = exports.S_IRUSR = 256;
const S_IWUSR = exports.S_IWUSR = 128;
const S_IXUSR = exports.S_IXUSR = 64;
const S_IRWXG = exports.S_IRWXG = 56;
const S_IRGRP = exports.S_IRGRP = 32;
const S_IWGRP = exports.S_IWGRP = 16;
const S_IXGRP = exports.S_IXGRP = 8;
const S_IRWXO = exports.S_IRWXO = 7;
const S_IROTH = exports.S_IROTH = 4;
const S_IWOTH = exports.S_IWOTH = 2;
const S_IXOTH = exports.S_IXOTH = 1;
const F_OK = exports.F_OK = 0;
const R_OK = exports.R_OK = 4;
const W_OK = exports.W_OK = 2;
const X_OK = exports.X_OK = 1;
const UV_FS_COPYFILE_EXCL = exports.UV_FS_COPYFILE_EXCL = 1;
const COPYFILE_EXCL = exports.COPYFILE_EXCL = 1;
const UV_FS_COPYFILE_FICLONE = exports.UV_FS_COPYFILE_FICLONE = 2;
const COPYFILE_FICLONE = exports.COPYFILE_FICLONE = 2;
const UV_FS_COPYFILE_FICLONE_FORCE = exports.UV_FS_COPYFILE_FICLONE_FORCE = 4;
const COPYFILE_FICLONE_FORCE = exports.COPYFILE_FICLONE_FORCE = 4;
const EXTENSIONLESS_FORMAT_JAVASCRIPT = exports.EXTENSIONLESS_FORMAT_JAVASCRIPT = 0;
const EXTENSIONLESS_FORMAT_WASM = exports.EXTENSIONLESS_FORMAT_WASM = 1;
const O_DIRECT = exports.O_DIRECT = 33554432;
const O_NOATIME = exports.O_NOATIME = 262144;
