// unenv/runtime/node/dns/internal/constants.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.V4MAPPED = exports.TIMEOUT = exports.SERVFAIL = exports.REFUSED = exports.NOTINITIALIZED = exports.NOTIMP = exports.NOTFOUND = exports.NONAME = exports.NOMEM = exports.NODATA = exports.LOADIPHLPAPI = exports.FORMERR = exports.FILE = exports.EOF = exports.DESTRUCTION = exports.CONNREFUSED = exports.CANCELLED = exports.BADSTR = exports.BADRESP = exports.BADQUERY = exports.BADNAME = exports.BADHINTS = exports.BADFLAGS = exports.BADFAMILY = exports.ALL = exports.ADDRGETNETWORKPARAMS = exports.ADDRCONFIG = void 0;
const ADDRCONFIG = exports.ADDRCONFIG = 0;
const ALL = exports.ALL = 0;
const V4MAPPED = exports.V4MAPPED = 2048;
const ADDRGETNETWORKPARAMS = exports.ADDRGETNETWORKPARAMS = "EADDRGETNETWORKPARAMS";
const BADFAMILY = exports.BADFAMILY = "EBADFAMILY";
const BADFLAGS = exports.BADFLAGS = "EBADFLAGS";
const BADHINTS = exports.BADHINTS = "EBADHINTS";
const BADNAME = exports.BADNAME = "EBADNAME";
const BADQUERY = exports.BADQUERY = "EBADQUERY";
const BADRESP = exports.BADRESP = "EBADRESP";
const BADSTR = exports.BADSTR = "EBADSTR";
const CANCELLED = exports.CANCELLED = "ECANCELLED";
const CONNREFUSED = exports.CONNREFUSED = "ECONNREFUSED";
const DESTRUCTION = exports.DESTRUCTION = "EDESTRUCTION";
const EOF = exports.EOF = "EEOF";
const FILE = exports.FILE = "EFILE";
const FORMERR = exports.FORMERR = "EFORMERR";
const LOADIPHLPAPI = exports.LOADIPHLPAPI = "ELOADIPHLPAPI";
const NODATA = exports.NODATA = "ENODATA";
const NOMEM = exports.NOMEM = "ENOMEM";
const NONAME = exports.NONAME = "ENONAME";
const NOTFOUND = exports.NOTFOUND = "ENOTFOUND";
const NOTIMP = exports.NOTIMP = "ENOTIMP";
const NOTINITIALIZED = exports.NOTINITIALIZED = "ENOTINITIALIZED";
const REFUSED = exports.REFUSED = "EREFUSED";
const SERVFAIL = exports.SERVFAIL = "ESERVFAIL";
const TIMEOUT = exports.TIMEOUT = "ETIMEOUT";
